src/main/java/net/openhft/chronicle/network/cluster/ClusterContext.java

```diff
@@ -357,12 +404,12 @@ protected void performClose() {
         acceptorLoop = null;
     }
 
-    private void closeAndWaitForEventLoops(@Nullable EventLoop... eventLoops) {
-        Arrays.stream(eventLoops)
+    private void closeAndWaitForEventLoops(@NotNull List<EventLoop> eventLoops) {
+        eventLoops.stream()
```

Comment:
don't we want just to stop them 1st then close during performClose?

Reply:
That's a good point... will fix

Reply:
For future reference, @JerryShea's suggestion was the right thing to do, but because there is no way to know when the handlers have stopped being called, after a call to `EventLoop#stop` we opted to stick with `EventLoop#close`. Perhaps it might be worth considering modifying the EventLoop contract so that `awaitTermination` works after a call to `EventLoop#stop`. But that is a larger piece of work.

Reply:
Yes. Another idea is that EventLoop.stop could block until loopFinished has been called